### PR TITLE
Add basic validation to environment parameters

### PR DESCRIPTION
**Pull request by krockpot**
The API currently does no validation of the parameters sent to the environment details page. This data is rendered back to users so we should do some validation to ensure scripts cannot be injected. 

I didn't see where our HTML templates get expanded, but am happy to move the HMTL escape to wherever we build/render the HTML for the user instead of storing the strings that way. Let me know!

### CODE REVIEW

**Comment by krockpot** [comment]
@yongwen @ankilosaurus for review

**Comment by krockpot** [comment]
@ankilosaurus  can you advise me on how to properly do the import here? I was just following the example in the environs.java file.

```
[ERROR] /home/travis/build/pinterest/teletraan/deploy-service/common/src/main/java/com/pinterest/deployservice/bean/EnvironBean.java:[18,41] package com.pinterest.teletraan.exception does not exist
[ERROR] /home/travis/build/pinterest/teletraan/deploy-service/common/src/main/java/com/pinterest/deployservice/bean/EnvironBean.java:[198,23] cannot find symbol
  symbol:   class TeletaanInternalException
  location: class com.pinterest.deployservice.bean.EnvironBean
```

**Comment by ankilosaurus** [comment]
> @ankilosaurus can you advise me on how to properly do the import here? I was just following the example in the environs.java file.
> 
> ```
> [ERROR] /home/travis/build/pinterest/teletraan/deploy-service/common/src/main/java/com/pinterest/deployservice/bean/EnvironBean.java:[18,41] package com.pinterest.teletraan.exception does not exist
> [ERROR] /home/travis/build/pinterest/teletraan/deploy-service/common/src/main/java/com/pinterest/deployservice/bean/EnvironBean.java:[198,23] cannot find symbol
>   symbol:   class TeletaanInternalException
>   location: class com.pinterest.deployservice.bean.EnvironBean
> ```

hi @krockpot, TeletraanInternalException is part of Teletraanservice module so you can not refer it from common module. How about use DeployInternalException?

File: deploy-service/common/src/main/java/com/pinterest/deployservice/bean/EnvironBean.java [hunk 1]
```diff
@@ -188,6 +191,32 @@
     @JsonProperty("ensureTrustedBuild")
     private Boolean ensure_trusted_build;
 
+    public void validate() throws Exception {
+        // A bunch of these fields will always be alphanumeric (with _ and -)
+        if(!this.env_name.matches("^[A-Za-z0-9_\\-]*$")) {
```
Comment (ankilosaurus) [inline deploy-service/common/src/main/java/com/pinterest/deployservice/bean/EnvironBean.java:196]: Hi Jeremy, Seems we might be adding lot of duplicate checks. Can we wrap it into @validate annotation and use instead for setters?

**Comment by krockpot** [comment]
API call performed with stage/environment names with non-alphanumeric characters results in:

```
No handlers could be found for logger "deploy_board.webapp.helpers.base_client"
Teletraan failed to call backend server. Contact your friendly Teletraan owners for assistance. Hint: 400, 
Message: The request entity had the following errors:
```

See attached image for UI result
Screen Shot 2020-08-07 at 1 55 58 PM

**Comment by krockpot** [comment]
abandoning for https://github.com/pinterest/teletraan/pull/741

**Pull request closed without merging**